pkg/types/validation/installconfig.go

```diff
@@ -699,3 +703,33 @@ func validateFIPSconfig(c *types.InstallConfig) field.ErrorList {
 	}
 	return allErrs
 }
+
+// validateCapabilities checks if additional, optional OpenShift components are specified in the
+// install-config to be included in the installation.
+func validateCapabilities(c *configv1.ClusterVersionCapabilitiesSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if _, found := configv1.ClusterVersionCapabilitySets[c.BaselineCapabilitySet]; !found {
+		allErrs = append(allErrs, field.Invalid(fldPath, c.BaselineCapabilitySet, "unknown baseline capability set specified"))
+	}
+
+	allAvailableCapabilities := map[string]bool{}
+	allCaps := []string{}
+	// Create a map and slice of *all* available capabilities across all capability sets
+	for k, _ := range configv1.ClusterVersionCapabilitySets {
+		for _, capability := range configv1.ClusterVersionCapabilitySets[k] {
+			if _, found := allAvailableCapabilities[string(capability)]; !found {
+				allAvailableCapabilities[string(capability)] = true
+				allCaps = append(allCaps, string(capability))
+			}
+		}
+	}
+	sort.Strings(allCaps)
```

Comment:
micro-nit: you can save fractions of CPU in the happy, error-free case by moving this sort down into the `!found` case below.  If we end up confirming the sort a few times when folks typo multiple capabilities, that seems like it would still be less energy than sorting all the runs where the user didn't typo anything.

Reply:
That makes sense. Thanks.